### ISSUE TITLE
Rework Run::current_split_index

### ISSUE THREAD

**Issue by CryZe** [LiveSplit/livesplit-core#266]
It should either return None or a valid index into the Run. I don't think it's a good idea that we are returning out of bounds indices here. @wooferzfg thoughts?

**Comment by wooferzfg** [comment]
Yeah, using an out of bounds index for when the run is unstarted or ended is really hacky because we can get the same information from the timer phase. We just need to make sure that there isn't any code that's relying on the current split index being an out of bounds value.

**Comment by CryZe** [comment]
I think we might even want to turn TimerPhase into:
```rust
enum AttemptState {
    Unstarted,
    Active {
        index: usize,
        paused: bool,
    },
    Ended,
}
```
or something similar.
